endpoint.go

```diff
@@ -243,15 +244,15 @@ func (ep *endpoint) KeyPrefix() []string {
 	return []string{datastore.EndpointKeyPrefix, string(n.id)}
 }
 
-func (ep *endpoint) networkIDFromKey(key []string) (types.UUID, error) {
-	// endpoint Key structure : endpoint/network-id/endpoint-id
-	// it's an invalid key if the key doesn't have all the 3 key elements above
-	if key == nil || len(key) < 3 || key[0] != datastore.EndpointKeyPrefix {
-		return types.UUID(""), fmt.Errorf("invalid endpoint key : %v", key)
+func (ep *endpoint) networkIDFromKey(key string) (string, error) {
+	// endpoint Key structure : docker/libnetwork/endpoint/${network-id}/${endpoint-id}
+	// it's an invalid key if the key doesn't have all the 5 key elements above
+	keyElements := strings.Split(key, "/")
+	if !strings.HasPrefix(key, "docker/libnetwork/endpoint/") || len(keyElements) < 5 {
```

Comment:
can you please replace `docker/libnetwork/endpoint/` with `datastore.Key(datastore.endpointKeyPrefix)` ?